src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -219,27 +272,28 @@ public void removeCitizen(final CitizenData citizen)
     {
         if (isWorker(citizen))
         {
-            setWorker(null);
+            citizen.setWorkBuilding(null);
+            workers.remove(citizen);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)